src/main/java/ch/njol/skript/expressions/ExprVehicle.java

```diff
@@ -38,102 +29,108 @@
-	
+
+	private static final boolean hasMountEvents = Skript.classExists("org.spigotmc.event.entity.EntityMountEvent");
+
 	static {
-		register(ExprVehicle.class, Entity.class, "vehicle[s]", "entities");
+		registerDefault(ExprVehicle.class, Entity.class, "vehicle[s]", "entities");
 	}
-	
+
 	@Override
-	protected Entity[] get(final Event e, final Entity[] source) {
+	protected Entity[] get(Event event, Entity[] source) {
 		return get(source, entity -> {
-			if (getTime() >= 0 && e instanceof VehicleEnterEvent && entity.equals(((VehicleEnterEvent) e).getEntered()) && !Delay.isDelayed(e)) {
-				return ((VehicleEnterEvent) e).getVehicle();
+			if (getTime() >= 0 && event instanceof VehicleEnterEvent && entity.equals(((VehicleEnterEvent) event).getEntered()) && !Delay.isDelayed(event)) {
+				return ((VehicleEnterEvent) event).getVehicle();
 			}
-			if (getTime() >= 0 && e instanceof VehicleExitEvent && entity.equals(((VehicleExitEvent) e).getExited()) && !Delay.isDelayed(e)) {
-				return ((VehicleExitEvent) e).getVehicle();
+			if (getTime() >= 0 && event instanceof VehicleExitEvent && entity.equals(((VehicleExitEvent) event).getExited()) && !Delay.isDelayed(event)) {
+				return ((VehicleExitEvent) event).getVehicle();
 			}
 			if (hasMountEvents) {
-				if (getTime() >= 0 && e instanceof EntityMountEvent && entity.equals(((EntityMountEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					return ((EntityMountEvent) e).getMount();
+				if (getTime() >= 0 && event instanceof EntityMountEvent && entity.equals(((EntityMountEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+					return ((EntityMountEvent) event).getMount();
 				}
-				if (getTime() >= 0 && e instanceof EntityDismountEvent && entity.equals(((EntityDismountEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					return ((EntityDismountEvent) e).getDismounted();
+				if (getTime() >= 0 && event instanceof EntityDismountEvent && entity.equals(((EntityDismountEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+					return ((EntityDismountEvent) event).getDismounted();
 				}
 			}
 			return entity.getVehicle();
 		});
 	}
-	
+
 	@Override
 	@Nullable
-	public Entity convert(final Entity e) {
+	public Entity convert(Entity entity) {
 		assert false;
-		return e.getVehicle();
-	}
-	
-	@Override
-	public Class<? extends Entity> getReturnType() {
-		return Entity.class;
-	}
-	
-	@Override
-	protected String getPropertyName() {
-		return "vehicle";
+		return entity.getVehicle();
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET) 
 			return new Class[] {Entity.class, EntityData.class};
-		}
 		return super.acceptChange(mode);
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
```

Comment:
Setting the vehicle to an entitydata in an on vehicle enter event seems like it's undefined behavior or something. Sometimes it puts me on the spawned entity properly, sometimes I don't get put on any vehicle, and sometimes I'm still in the boat but can't dismount, and relogging puts me back at the start, which indicates client-server desync to me. Perhaps it'd be best to just forbid setting it in the mount/enter event, since I can't really think of any specific cause or fix for this.